pkg/providers/smi/provider.go

```diff
@@ -162,6 +191,27 @@ func (p *Provider) BuildConfig() (*dynamic.Configuration, error) {
 	return config, nil
 }
 
+func (p *Provider) getTCPPort(namespace, name string, port int32) (int32, error) {
```

Comment:
since this is required for multiple providers could it be moved to the base?

That way the kubernetes and SMI providers could use it.

Reply:
`getTCPPort` and `getHTTPPort` are not helpers, they both relies on the Provider' state.